soccer/planning/RRTPlanner.hpp

```diff
@@ -71,25 +89,54 @@ class RRTPlanner : public SingleRobotPathPlanner {
                       const Path* prevPath) const;
 
     /// Runs a bi-directional RRT to attempt to join the start and end states.
-    Planning::InterpolatedPath* runRRT(
+    std::unique_ptr<Planning::InterpolatedPath> runRRT(
         MotionInstant start, MotionInstant goal,
         const MotionConstraints& motionConstraints,
-        const Geometry2d::ShapeSet* obstacles);
+        const Geometry2d::ShapeSet* obstacles, int smoothingIterations = 0);
 
     /** optimize the path
      *  Calls the cubicBezier optimization function.
      */
-    Planning::InterpolatedPath* optimize(
-        Planning::InterpolatedPath& path, const Geometry2d::ShapeSet* obstacles,
+    std::unique_ptr<Planning::InterpolatedPath> optimize(
+        std::unique_ptr<InterpolatedPath> path,
+        const Geometry2d::ShapeSet* obstacles,
         const MotionConstraints& motionConstraints, Geometry2d::Point vi,
-        Geometry2d::Point vf);
+        Geometry2d::Point vf, int smoothingIterations = 0);
 
     /**
      * Uses a cubicBezier to interpolate between the points on the path and add
      * velocity planning
      */
-    static Planning::InterpolatedPath* cubicBezier(
-        Planning::InterpolatedPath& path, const Geometry2d::ShapeSet* obstacles,
+    static std::unique_ptr<Planning::InterpolatedPath> cubicBezier(
+        std::unique_ptr<Planning::InterpolatedPath> path,
+        const Geometry2d::ShapeSet* obstacles,
+        const MotionConstraints& motionConstraints, Geometry2d::Point vi,
+        Geometry2d::Point vf);
+
+    /**
+     * Generates a Cubic Bezier Path based on Albert's random Bezier Velocity
```

Comment:
What does the "random" refer to here?  Is this just because it's smoothing a path that's originally from an RRT?